src/xdp/trn_transit_xdp.c

```diff
@@ -175,44 +176,49 @@ static __inline int trn_process_inner_ip(struct transit_packet *pkt)
 		flow->dport = pkt->inner_udp->dest;
 	}
 
-	/* Modify inner EtherHdr */
-	trn_set_dst_mac(pkt->inner_eth, ep->mac);
-
-	/* Keep overlay header, update outer header destinations */
-	trn_set_src_dst_ip_csum(pkt->ip, pkt->ip->saddr, ep->hip, pkt->data_end);
-	trn_set_dst_mac(pkt->eth, ep->hmac);
-
 	/* Generate Direct Path request */
-	pkt->fctx.opcode = XDP_FLOW_OP_ENCAP;
+	pkt->fctx.opcode = bpf_htonl(XDP_FLOW_OP_ENCAP);
 	pkt->fctx.opdata.encap.dip = epkey.ip;
 	pkt->fctx.opdata.encap.dhip = ep->hip;
 	trn_set_mac(pkt->fctx.opdata.encap.dmac, ep->mac);
 	trn_set_mac(pkt->fctx.opdata.encap.dhmac, ep->hmac);
-	pkt->fctx.opdata.encap.timeout = TRAN_DP_FLOW_TIMEOUT;
+	pkt->fctx.opdata.encap.timeout = bpf_htons(TRAN_DP_FLOW_TIMEOUT);
 
 	pkt->fctx.udp.dest = pkt->itf->ibo_port;
-	pkt->fctx.udp.len = sizeof(struct udphdr) + sizeof(pkt->fctx.opcode) +
+	len = sizeof(struct udphdr) + sizeof(pkt->fctx.opcode) +
 		sizeof(pkt->fctx.flow) + sizeof(dp_encap_opdata_t);
+	pkt->fctx.udp.len = bpf_htons(len);
 
 	pkt->fctx.ip.version = IPVERSION;
 	pkt->fctx.ip.ihl = sizeof(struct iphdr) >> 2;
-	pkt->fctx.ip.tot_len = sizeof(struct iphdr) + pkt->fctx.udp.len;
+	len += sizeof(struct iphdr);
+	pkt->fctx.ip.tot_len = bpf_htons(len);
```

Comment:
Should we change this value to network byte order then add? Instead of add then swap byte order. Or does it not matter?

Reply:
No, the add is in host context, the result is put in buffer in network order, this is correct. Adding (by value) in swapped order would get you wrong result actually if there carry bit.